test/modules/TestResultCoordinator/Reports/EdgeHubRestartTest/EdgeHubRestartDirectMethodReportGenerator.cs

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace TestResultCoordinator.Reports.EdgeHubRestartTest
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil.TestResults;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    sealed class EdgeHubRestartDirectMethodReportGenerator : ITestResultReportGenerator
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger(nameof(EdgeHubRestartDirectMethodReportGenerator));
+
+        internal EdgeHubRestartDirectMethodReportGenerator(
+            string trackingId,
+            string senderSource,
+            string receiverSource,
+            TestReportType testReportType,
+            ITestResultCollection<TestOperationResult> senderTestResults,
+            ITestResultCollection<TestOperationResult> receiverTestResults,
+            TimeSpan passableEdgeHubRestartPeriod)
```

Comment:
Please explain why do we need this.  Current timeout should be good enough to decide whether test run is passed or not.

Reply:
This is for direct method which include the SDK connection time for the receiver module reconnection to edgeHub.

Reply:
So report generator will calculate time duration between edgehub restart timestamp and receiver timestamp; then if  time duration is longer then "passableEdgeHubRestartPeriod", it is failed; otherwise passed. do I understand correctly?  I think this is not worth to do; @varunpuranik what do you think?